### PR TITLE
track dapp suggested gas fees

### PR DESCRIPTION
**Pull request by brad-decker**
New designs/implementation will need to be fully aware of the original values submitted in a tx by the dapp. This adds those values to the txMeta so we can reference them in future work.

### CODE REVIEW

**Review by NiranjanaBinoy — DISMISSED** [review]
LGTM

File: app/scripts/controllers/transactions/index.js [hunk 1]
```diff
@@ -300,6 +300,8 @@ export default class TransactionController extends EventEmitter {
       txParams: normalizedTxParams,
     });
 
+    txMeta.dappSuggestedGasFees = null;
```
Comment (darkwing) [inline app/scripts/controllers/transactions/index.js:303]: Do we want to add `dappSuggestedGasFees` to what `generateTxMeta` returns?

Comment (brad-decker) [reply app/scripts/controllers/transactions/index.js]: Smart move. DId that and added some tests.

File: app/scripts/controllers/transactions/tx-state-manager.test.js [hunk 2]
```diff
@@ -1074,6 +1075,136 @@ describe('TransactionStateManager', function () {
     });
   });
 
+  describe('#generateTxMeta', function () {
+    it('generates a txMeta object when supplied no parameters', function () {
+      // There are currently not safety checks for missing 'opts' but we should
+      // at least enforce txParams. This is done in the transaction controller
+      // before *calling* this method, but we should perhaps ensure that
+      // txParams is provided and validated in this method.
+      // TODO: this test should fail.
```
Comment (darkwing) [inline app/scripts/controllers/transactions/tx-state-manager.test.js:1084]: Is this comment relevant?  Should we be making it fail now?

Comment (brad-decker) [reply app/scripts/controllers/transactions/tx-state-manager.test.js:1084]: Yes, this comment was meant to be leaving a TODO for myself as I explore optimizations / improvements to the tx-state-manager/transaction controller improvements.

It wasn't meant to be made to fail in this PR.

**Review by darkwing — APPROVED** [review]

**Review by NiranjanaBinoy — APPROVED** [review]
LGTM

**Pull request merged**